pkg/controllers/user/serviceaccount/serviceaccount.go

```diff
@@ -0,0 +1,343 @@
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"strings"
+
+	v13 "github.com/rancher/types/apis/core/v1"
+	v12 "github.com/rancher/types/apis/rbac.authorization.k8s.io/v1"
+	v1beta12 "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const apiGroup = "rbac.authorization.k8s.io"
+const apiVersion = "extensions/v1beta1"
+
+const podSecurityAnnotation = "serviceaccount.cluster.cattle.io/pod-security"
+const projectIdAnnotation = "field.cattle.io/projectId"
+const podSecurityPolicyTemplateLabel = "service.cluster.cattle.io/pod-security-policy-template-id"
+
+func Register(context *config.UserContext) {
+	logrus.Infof("registering serviceaccount handler for cluster %v", context.ClusterName)
+
+	m := &manager{
+		clustersInt:  context.Management.Management.Clusters(""),
+		clusters:     context.Management.Management.Clusters("").Controller().Lister(),
+		templates:    context.Management.Management.PodSecurityPolicyTemplates("").Controller().Lister(),
+		listPolicies: context.Extensions.PodSecurityPolicies("").Controller().Lister(),
+		listBindings: context.RBAC.ClusterRoleBindings("").Controller().Lister(),
+		bindings:     context.RBAC.ClusterRoleBindings(""),
+		policies:     context.Extensions.PodSecurityPolicies(""),
+		listRoles:    context.RBAC.ClusterRoles("").Controller().Lister(),
+		roles:        context.RBAC.ClusterRoles(""),
+		namespaces:   context.Core.Namespaces("").Controller().Lister(),
+		projects:     context.Management.Management.Projects("").Controller().Lister(),
+	}
+
+	context.Core.ServiceAccounts("").AddHandler(
+		"universalServiceAccountHandler",
+		func(key string, obj *v1.ServiceAccount) error {
+			logrus.Info("handle service account")
+
+			if obj == nil {
+				logrus.Infof("no service account provided, exiting")
+				// do nothing
+				return nil
+			}
+
+			namespaceName := obj.Namespace
+
+			logrus.Infof("svc acct name: %v namespace: %v", obj.Name, namespaceName)
+
+			// Check if a PSP exists for this cluster
+			policies, err := m.listPolicies.List("", labels.Everything())
+
+			if err != nil {
+				return fmt.Errorf("error getting policies: %v", err)
+			}
+
+			bindings, err := m.listBindings.List(namespaceName, labels.Everything())
+
+			if err != nil {
+				return fmt.Errorf("error getting cluster role bindings: %v", err)
+			}
+
+			var policy *v1beta12.PodSecurityPolicy
+
+			for _, candidate := range policies {
+				if candidate.ObjectMeta.Annotations[podSecurityAnnotation] == "true" {
+					policy = candidate
+				}
+			}
+
+			if policy == nil {
+				// get PSPT
+				// if no PSPT then get default
+				// if no default then exit
+				namespaces, err := m.namespaces.List("", labels.Everything())
+
+				if err != nil {
+					return fmt.Errorf("error getting namespaces: %v", err)
+				}
+
+				var namespace *v1.Namespace
+
+				for _, candidate := range namespaces {
```

Comment:
just use m.namespaceLister.Get(namespaceName, ...)
no need to loop

Reply:
this applies to other places in the code too, but will just leave this one comment